Flank/src/main/java/com/walmart/otto/Flank.java

```diff
@@ -233,21 +256,29 @@ private void printShards(Configurator configurator, int numberOfShards) {
 
   private List<String> getTestCaseNames(OptionSet options) {
     System.setOut(getEmptyStream());
-    List<String> filteredTests = new ArrayList<>();
+    List<TestMethod> testMethods = DexParser.findTestMethods((String) options.valueOf("t"));
+    System.setOut(originalStream);
 
-    for (TestMethod testMethod : DexParser.findTestMethods((String) options.valueOf("t"))) {
-      if (!testMethod.getAnnotationNames().stream().anyMatch(str -> str.contains("Ignore"))) {
-        filteredTests.add(testMethod.getTestName());
-      }
+    final TestFilter filter;
+    if (options.has("f")) {
+      filter = TestFilters.fromCommandLineArguments((String) options.valueOf("f"));
+    } else {
+      filter = TestFilters.createDefault();
     }
 
-    if (options.has("p")) {
-      filteredTests =
-          FilterUtils.filterTests(
-              filteredTests, configurator.getSkipTests(), (String) options.valueOf("p"));
+    List<String> filteredTests =
+        testMethods
+            .stream()
+            .filter(filter::shouldRun)
+            .map(TestMethod::getTestName)
+            .collect(Collectors.toList());
+
+    if (filteredTests.size() == 0) {
+      throw new IllegalStateException("No tests to run!");
     }
 
-    System.setOut(originalStream);
+    System.out.printf("Running %d out of %d tests", filteredTests.size(), testMethods.size());
```

Comment:
Aren't there logs in the Flank class which prints similar info?

Reply:
Not really, I think we print just the number of shards.